native/explorer/src/datatypes.rs

```diff
@@ -380,7 +380,37 @@ fn encode_datetime_series<'b>(s: &Series, time_unit: TimeUnit, env: Env<'b>) ->
                     .as_c_arg()
             })
         })
-        .collect::<Vec<NIF_TERM>>();
+        .collect();
+
+    unsafe { Term::new(env, make_list(env.as_c_arg(), &items)) }
+}
+
+// Convert f64 series taking into account NaN and Infinity floats (they are encoded as atoms).
+fn encode_float64_series<'b>(s: &Series, env: Env<'b>) -> Term<'b> {
+    let items: Vec<NIF_TERM> = s
+        .f64()
+        .unwrap()
+        .downcast_iter()
+        .flat_map(move |iter| {
+            iter.into_iter().map(move |opt_v| {
+                opt_v
+                    .copied()
+                    .map(|x| {
+                        if x.is_finite() {
+                            x.encode(env)
+                        } else {
+                            match (x.is_nan(), x.is_sign_negative()) {
```

Comment:
If you put the `nan`, `neg_infinity`, `infinity` encode part at the top of the function so it doesn't have to encode each time, as these don't change.

Reply:
Oh, nice! I changed the code to fix that. Thank you!